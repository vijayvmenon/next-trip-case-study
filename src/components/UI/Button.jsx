import React from 'react';
import PropTypes from 'prop-types';
import { createUseStyles } from 'react-jss';
import { colors } from '../../constants';

const useStyles = createUseStyles({
  root: {
    background: colors.BACKGROUND_2,
    color: colors.WHITE,
    borderRadius: 5,
    fontSize: '1rem',
    fontWeight: 'bold',
    padding: '0px 16px',
    height: 40,
    cursor: 'pointer',
  },
});

export default function Button({ text, click }) {
  const classes = useStyles();
  return (
    <button
      type="button"
      className={classes.root}
      onClick={click}
      data-testid="custom-button"
    >
      {text}
    </button>
  );
}

Button.defaultProps = {
  text: '',
};

Button.propTypes = {
  text: PropTypes.string,
  click: PropTypes.func.isRequired,
};
