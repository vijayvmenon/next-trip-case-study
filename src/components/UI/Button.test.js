import { render, fireEvent, screen } from '@testing-library/react';
import Button from './Button';
import renderer from 'react-test-renderer';

const props = {
  text: 'test',
  click: jest.fn(),
};

describe('Component:Button.jsx', () => {
  test('Button.jsx renders correctly', () => {
    const button = renderer.create(<Button {...props} />).toJSON();
    expect(button).toMatchSnapshot();
  });

  test('Buttons click prop is called onClick', () => {
    const { getByTestId } = render(<Button {...props} />);
    fireEvent.click(getByTestId('custom-button'));
    expect(props.click).toHaveBeenCalledTimes(1);
  });
});
