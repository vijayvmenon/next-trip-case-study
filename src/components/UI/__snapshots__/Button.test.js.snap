// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component:Button.jsx Button.jsx renders correctly 1`] = `
<button
  className="root-0-2-1"
  data-testid="custom-button"
  onClick={[MockFunction]}
  type="button"
>
  test
</button>
`;
